common/logger.go

```diff
@@ -159,6 +160,14 @@ func (jl *jobLogger) ShouldLog(level pipeline.LogLevel) bool {
 	return level <= jl.minimumLevelToLog
 }
 
+func (jl *jobLogger) ChangeLogLevel(level pipeline.LogLevel) bool {
+	if level == pipeline.LogNone {
+		return false
+	}
+	jl.minimumLevelToLog = level
```

Comment:
Now that we are writing to this variable, we can have multiple goroutines call this and technically have issues.
Though we may be calling it only once, but better to call out.

// This update is not necessarily safe from multiple goroutines simultaneously calling it.
// Typically we will call ChangeLogLevel() once at the beginning so it should be ok.